extensions/banner/runtime/src/main/java/io/quarkus/banner/runtime/BannerRecorder.java

```diff
@@ -0,0 +1,36 @@
+package io.quarkus.banner.runtime;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import io.quarkus.deployment.util.FileUtil;
+import io.quarkus.runtime.RuntimeValue;
+import io.quarkus.runtime.annotations.Recorder;
+
+@Recorder
+public class BannerRecorder {
+    public RuntimeValue<Optional<Supplier<String>>> provideBannerSupplier(BannerConfig config) {
+        String banner = readBannerFile(config);
+        Supplier<String> supplierValue = () -> banner;
```

Comment:
This supplier should do the actual banner file reading, so that the banner is only held in memory when it is actually being displayed.

Reply:
So I would pass the path of banner.txt to supplier. And supplier will read it from the path when it provides string. ?

Reply:
Done